README.md

```diff
@@ -32,13 +34,113 @@ Go 1.5 users will need to set `GO15VENDOREXPERIMENT=1` to get vendored dependenc
+
+Use Mellanox Firmware Tools package to enable and configure SR-IOV in firmware
+
+```
+# mst start
+Starting MST (Mellanox Software Tools) driver set
+Loading MST PCI module - Success
+Loading MST PCI configuration module - Success
+Create devices
+```
+
+Locate the HCA device on the desired PCI slot
+
+```
+# mst status
+MST modules:
+------------
+    MST PCI module loaded
+    MST PCI configuration module loaded
+MST devices:
+------------
+/dev/mst/mt4115_pciconf0         - PCI configuration cycles access.
+...
+```
+
+Enable SR-IOV
+
+```
+# mlxconfig -d /dev/mst/mt4115_pciconf0 q set SRIOV_EN=1 NUM_OF_VFS=8
+...
+Apply new Configuration? ? (y/n) [n] : y
+Applying... Done!
+-I- Please reboot machine to load new configurations.
+```
+
+Reboot the machine
+```
+# reboot
+```
+
+3- Enable SR-IOV in the NIC's Driver.
+
+```
+# ibdev2netdev
+mlx5_0 port 1 ==> enp2s0f0 (Up)
+mlx5_1 port 1 ==> enp2s0f1 (Up)
+
+# echo 4 > /sys/class/net/enp2s0f0/device/sriov_numvfs
+# ibdev2netdev -v
+0000:02:00.0 mlx5_0 (MT4115 - MT1523X04353) CX456A - ConnectX-4 QSFP fw 12.23.1020 port 1 (ACTIVE) ==> enp2s0f0 (Up)
+0000:02:00.1 mlx5_1 (MT4115 - MT1523X04353) CX456A - ConnectX-4 QSFP fw 12.23.1020 port 1 (ACTIVE) ==> enp2s0f1 (Up)
+0000:02:00.5 mlx5_2 (MT4116 - NA)  fw 12.23.1020 port 1 (DOWN  ) ==> enp2s0f2 (Down)
+0000:02:00.6 mlx5_3 (MT4116 - NA)  fw 12.23.1020 port 1 (DOWN  ) ==> enp2s0f3 (Down)
+0000:02:00.7 mlx5_4 (MT4116 - NA)  fw 12.23.1020 port 1 (DOWN  ) ==> enp2s0f4 (Down)
+0000:02:00.2 mlx5_5 (MT4116 - NA)  fw 12.23.1020 port 1 (DOWN  ) ==> enp2s0f5 (Down)
+
+# lspci | grep Mellanox
+02:00.0 Ethernet controller: Mellanox Technologies MT27700 Family [ConnectX-4]
+02:00.1 Ethernet controller: Mellanox Technologies MT27700 Family [ConnectX-4]
+02:00.2 Ethernet controller: Mellanox Technologies MT27700 Family [ConnectX-4 Virtual Function]
+02:00.3 Ethernet controller: Mellanox Technologies MT27700 Family [ConnectX-4 Virtual Function]
+02:00.4 Ethernet controller: Mellanox Technologies MT27700 Family [ConnectX-4 Virtual Function]
+02:00.5 Ethernet controller: Mellanox Technologies MT27700 Family [ConnectX-4 Virtual Function]
+
+# ip link show
+...
+enp2s0f2: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
+    link/ether c6:6d:7d:dd:2a:d5 brd ff:ff:ff:ff:ff:ff
+enp2s0f3: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
+    link/ether 42:3e:07:68:da:fb brd ff:ff:ff:ff:ff:ff
+enp2s0f4: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
+    link/ether 42:68:f2:aa:c2:27 brd ff:ff:ff:ff:ff:ff
+enp2s0f5: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
+...
+```
+
+To change the number of VFs reset the number to 0 then set the needed number
+
+```
+echo 0 > /sys/class/net/enp2s0f0/device/sriov_numvfs
+echo 8 > /sys/class/net/enp2s0f0/device/sriov_numvfs
+```
```

Comment:
do we have an working example for dpdk configuration? for example, which ``kernel_driver`` and ``dpdk_driver`` shall be used?  This might fall into ``Usage - Configuration with DPDK`` section.

Reply:
Plus any special cases need to be considerred when using Mellanox cards?

Reply:
no

Reply:
sure, then we might want to add a note that dpdk is not yet tested with Mellanox NIC, then we update it once the test is done. or perhaps wait until the test be executed then merge this PR?

Reply:
so I test DPDK, for Mellanox we don't need the DPDK config options as we are using the kernel driver (it is  Bifurcated Linux Driver  see https://doc.dpdk.org/guides/nics/mlx5.html). To make dpdk work we need first /dev/infiniband (RDMA),  move the VF to the container namespace  and adding     securityContext:      capabilities:        add: ["IPC_LOCK", "NET_ADMIN", "NET_RAW"]. 
I am trying to see if I can find a workaround to mount the /dev/infiniband for the pod yaml file. If it won't work I guess we can say that dpdk is not support yet.

Reply:
so we can mount the /dev/infiniband in the yaml of the container so we can say that is working, So for now merged this as is and @Mmduh-483  will provide a follow-up patch for the DPDK support. Is that ok with you?

Reply:
@zshi-redhat, can we merge this?